miden-lib/asm/wallets/basic.masm

```diff
@@ -0,0 +1,28 @@
+use.miden::sat::account
+use.miden::sat::tx
+
+#! This procedure adds the given asset to the undelying account.
+#! 
+#! Inputs: [ASSET]
+#! Outputs: [0, 0, 0, 0, ...]
+#!
+#! - ASSET is the asset to be received, can be fungible or non-fungible
+export.receive_asset
+    exec.account::add_asset
+    dropw
+end
```

Comment:
I think this will drop the asset from the stack but won't push zeros onto it. We should probably change this to:
```
export.receive_asset
    exec.account::add_asset
    dropw padw
end
```

Reply:
First, I thought so too, but when I add a `padw` or `push.0.0.0.0` I get 

```
ExecuteTransactionProgramFailed(InvalidStackDepthOnReturn(20))
```
___
**Potential reason:** 

Maybe `add_asset` in `/asm/sat/account.masm` doesn't work as expected.

```
#! Add the specified asset to the vault.
#!
#! Panics:
#! - If the asset is not valid.
#! - If the total value of two fungible assets is greater than or equal to 2^63.
#! - If the vault already contains the same non-fungible asset.
#!
#! Stack: [ASSET]
#! Output: [ASSET']
#!
#! - ASSET' final asset in the account vault defined as follows:
#!   - If ASSET is a non-fungible asset, then ASSET' is the same as ASSET.
#!   - If ASSET is a fungible asset, then ASSET' is the total fungible asset in the account vault
#!     after ASSET was added to it.
export.add_asset
    syscall.account_vault_add_asset
end
```

But here `syscall.account_vault_add_asset` I am not sure which function that is in internal

Reply:
For future debuggin, we should have a way to show the stack after a certain script / procedure is called. Maybe that is already doable if I run it in debug mode and add a breakpoint.

Reply:
Not doable yet, but something I proposed in https://github.com/0xPolygonMiden/miden-vm/issues/988#issuecomment-1711099264 shouldn't be too difficult to add.

Reply:
ok, I know what's going on (in hindsight, should have been obvious). If we do just `dropw padw` as I suggested, we actually would get a wrong result. Here is how stack transitions would work in this case:
```
# => [ASSET, X, Y, Z]

dropw
# => [X, Y, Z, 0, 0, 0, 0]
# since we drop the top 4 elements, and the depth of the stack must remain 16
# 4 zeros are shifted in from the right

padw
# => [0, 0, 0, 0, X, Y, Z, 0, 0, 0, 0]
# now the depth of the stack is 20 - but to return from the call it must be exactly 16
```

A simple solution to fix this would be:
```
export.receive_asset
    exec.account::add_asset
    padw swapw dropw
end
```

Reply:
Looks right, thanks for the analysis